deepchem/applications/synthesis.py

```diff
@@ -0,0 +1,185 @@
+from typing import Tuple, Dict, Any, Union, List
+
+_SettingsType = Dict[str, Union[int, float, str, bool]]
+
+
+class RetrosynthesisEngine:
+  """Retrosynthesis engine
+
+  This holds an interface to the AiZynthFinder retrosynthesis library
+
+  References
+  ----------
+  Genheden S, Thakkar A, Chadimova V, et al (2020) J. Cheminf. 12:70, 10.1186/s13321-020-00472-1
+
+  Note
+  ----
+  This class requires AiZynthFinder to be installed.
+  """
+
+  _policy_name = "deepchem_policy"
+  _stock_name = "deepchem_stock"
+
+  def __init__(self,
+               expansion_policy_paths: Tuple[str, str],
+               stock_path: str,
+               filter_policy_path: str = None) -> None:
+    """Initialize the engine
+
+    Parameters
+    ----------
+    expansion_policy_paths: tuple of str,
+      the path to the expansion policy model and the corresponding template file
+    stock_path: str
+      the path to the file with the stock molecules as InChI keys
```

Comment:
Could you add a pointer that explains what stock molecules are? A pointer to the aizynthfinder docs or an explanation here of the desired format would be very useful